src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -41,6 +43,35 @@ public void variousFormatReaderTest(final String inputFile) throws IOException {
         reader.close();
     }
 
+    @Test(dataProvider = "variousFormatReaderTestCases")
```

Comment:
This seems like the wrong `DataProvider` to use with this test case, since the `DataProvider` has only one bam and a bunch of sams, and the inflater functionality affects only the bam. Perhaps just use the bam from that provider directly in this test case.